Feat/customer domain

jruiz002 opened the pull request:
✨ Descripción
Este Pull Request introduce la implementación completa del flujo de eliminación de ítems del wishlist, incluyendo la integración de eventos de dominio para garantizar la consistencia entre capas y la correcta propagación de cambios.

Problema Resuelto / Característica Implementada:
Implementa la eliminación individual y múltiple de ítems en el wishlist con manejo de eventos de dominio.
Mejora la trazabilidad de las operaciones mediante event handlers asociados a las mutaciones.
Refactoriza los servicios de wishlist para adaptarlos a la arquitectura DDD y CQRS.

Impacto:
Afecta directamente a las operaciones de mutación GraphQL relacionadas con el wishlist.
Posible impacto en integridad si no se sincronizan correctamente los eventos de eliminación, especialmente en entornos concurrentes.

🚀 Cambios
Implementación de deleteWishlistItem mutation con su flujo de comando y manejador (DeleteWishlistItemCommandHandler).
Creación de DeleteWishlistItemEvent y sus correspondientes event handlers para manejar la lógica post-eliminación.
Implementación de bulkDeleteWishlistItems mutation para soportar eliminación múltiple de ítems.
Añadido soporte para event handling en el dominio wishlist, siguiendo los principios de Domain-Driven Design.
Actualización del WishlistRepository con nuevos métodos de eliminación por variantId y múltiples variantes.
Adaptación del WishlistService y WishlistAdapter para incluir detalles de variante en las respuestas.
Limpieza y unificación de las pruebas existentes para asegurar la cobertura de los nuevos casos de eliminación.

✅ Checklist

- [x]  He leído la guía [CONTRIBUTING.md](https://github.com/GerardoFdez7/easystore-services/blob/main/CONTRIBUTING.md)
- [x]  He realizado una auto-revisión de mi propio código.
- [x]  He comentado mi código, particularmente en áreas difíciles de entender.
- [x]  Mis cambios no generan nuevas advertencias.
- [x]  Todas las pruebas unitarias/de integración existentes y nuevas pasan localmente con mis cambios.
- [x]  Se ha respetado la Arquitectura DDD (Domain, Application, Infrastructure, Presentation) en la implementación.
- [x]  Los patrones CQRS, Factory y Repository han sido aplicados correctamente.
- [x]  Las transacciones y el manejo de errores han sido considerados y probados.

jruiz002 committed:
feat(customer): implement customer domain with entities, events and mappers

- Add customer domain module with entities, events, mappers and repositories
- Implement customer creation flow with commands and events
- Update authentication domain to handle customer creation on registration
- Remove duplicate customer-created event from cart domain
- Add GraphQL types and resolver for customer operations

jruiz002 committed:
refactor(customer): rename event handler and remove debug logs

Rename CustomerCreatedHandler to IdentityCreatedHandler to better reflect its purpose
Remove console.log statements used for debugging

jruiz002 committed:
refactor(authentication): move customer repository to customer domain

Move customer repository and related interfaces from authentication domain to customer domain to better align with domain boundaries. This includes updating imports and removing obsolete files from authentication domain.

jruiz002 committed:
feat(customer): add CustomerCreatedHandler to event handlers

Register CustomerCreatedHandler to handle customer creation events

jruiz002 committed:
feat(tenant): add domain-based tenant lookup and module exports

- Export ITenantRepository from tenant module for cross-domain usage
- Implement getTenantIdByDomain method in tenant repository
- Update customer module to use tenant domain for tenant ID resolution
- Reorder imports in app module for consistency

jruiz002 committed:
refactor(customer): use full domain and email for tenant lookup

Previously used the first part of domain for tenant lookup, now using full domain and extracting name from email instead for more accurate tenant identification

jruiz002 committed:
feat(customer): add find customer by id query

Implement query to find customer by ID including:
- New query handler, DTO and resolver
- Added repository method and interface
- Updated module to register query handler
- Added GraphQL schema type and query

jruiz002 committed:
feat(customer): add customer update functionality

implement customer update feature including event, handler, repository method and graphql resolver

jruiz002 committed:
feat(customer): add wishlist functionality

- Implement wishlist value objects, entities and repositories
- Add GraphQL types and resolver for wishlist operations
- Create wishlist event handlers and command handlers
- Add wishlist related DTOs and mappers

jruiz002 committed:
feat(wishlist): add delete wishlist item functionality

Implement wishlist item deletion by adding new DTO, repository method, and command handler

jruiz002 committed:
feat(wishlist): implement wishlist item deletion with event handling

- Add WishlistItemDeletedEvent and handler
- Update repository to return deleted wishlist item
- Modify delete command handler to emit event with deleted item

jruiz002 committed:
feat(wishlist): add mutation to remove variant from wishlist

Implement mutation endpoint to allow customers to remove items from their wishlist

jruiz002 committed:
feat(wishlist): add bulk deletion for wishlist items

implement functionality to delete multiple wishlist items at once
add new DTO, event, repository method and handler
update customer module with new components

jruiz002 committed:
feat(wishlist): add mutation to remove multiple variants from wishlist

Implement new mutation to allow bulk deletion of wishlist items by variant IDs. This improves efficiency when users need to remove multiple items at once.

jruiz002 committed:
feat(customer): add product adapter and port for variant details

Implement product adapter to fetch variant details using CQRS pattern

jruiz002 committed:
feat(wish-list): add method to get wish lists by variant ids

Implement getManyWishListsByVariantIds to fetch multiple wish list items by variant IDs and customer ID

jruiz002 committed:
feat(wishlist): add enriched wishlist items query with variant details

- Introduce new query handler to fetch wishlist items with product variant details
- Add WishlistItemWithVariantDTO and corresponding GraphQL types
- Reorganize customer query files into more specific directories
- Update GraphQL schema and resolver to support new query

jruiz002 committed:
feat(wishlist): add pagination support for wishlist items query

implement pagination for wishlist items query including new DTOs, input type and resolver changes

jruiz002 committed:
feat(customer): add customer review product value object

Implement value object for customer product reviews with validation schema, persistence handling, and getter methods

jruiz002 committed:
feat(customer): add customer review product functionality

- Add new interfaces, events and repository for customer reviews
- Implement review creation in customer entity
- Register repository in customer module

jruiz002 committed:
feat(customer): add customer review product DTO and mapper

Implement DTO interfaces and mapper class for customer product reviews to handle data transfer between layers. Includes methods for persistence conversion and pagination mapping.

jruiz002 committed:
feat(customer): add customer review product creation handler

Implement handler for creating customer product reviews with DTO and repository integration

jruiz002 committed:
feat(customer): add review creation command and event handlers

Implement command handler for creating customer product reviews and corresponding event handler
Remove duplicate handler file and register new handlers in customer module

jruiz002 committed:
feat(customer): add product review functionality

- Add CustomerReviewProductType and CreateCustomerReviewProductInput types
- Update GraphQL schema with review-related types and mutation
- Implement addReviewProduct resolver to handle product reviews

jruiz002 committed:
refactor(value-objects): update zod import and use uuidv7 validation

Update zod import from 'zod' to 'zod/v4' for better version management.
Replace string uuid validation with uuidv7 which provides better error messages.

jruiz002 committed:
feat(customer): add review update functionality

implement review update method in CustomerReviewProduct value object
add CustomerReviewProductUpdatedEvent and export it
extend Customer entity with updateCustomerReviewProduct method

jruiz002 committed:
feat(customer-review): add update method to customer review repository

Implement update functionality for customer reviews including error handling for database operations

jruiz002 committed:
feat(review): add update customer review product functionality

- Implement update review DTO, handler and event
- Add findById method to review repository
- Update GraphQL schema and resolver with new mutation
- Reorganize customer update command files into subdirectories

jruiz002 committed:
feat(customer-review): add removeReview method to repository interface and implementation

Implement removeReview method to allow deleting customer reviews by customer and review IDs
Handle cases where review doesn't exist or doesn't belong to customer

jruiz002 committed:
feat(customer): add delete customer review product functionality

implement command handler, DTO, event and entity method for deleting customer reviews

jruiz002 committed:
feat(customer): add delete review product mutation

- Add DeleteCustomerReviewProductInput type
- Register DeleteCustomerReviewProductHandler in module
- Add deleteReviewProduct mutation to schema
- Implement deleteReviewProduct resolver method

jruiz002 committed:
feat(customer): add findMany method to customer review product repository

Implement findMany method to retrieve multiple customer review products by customer ID and optional review IDs. This enables more efficient batch fetching of reviews when needed.

jruiz002 committed:
feat(customer): add paginated customer reviews query with variant details

implement getCustomerReviews query to fetch paginated reviews with enriched product variant information

jruiz002 committed:
feat(customer): add update customer mutation to GraphQL schema and resolver

Implement the update customer functionality allowing customers to modify their name and default addresses/phone numbers

GerardoFdez7 commented on src/domains/customer/aggregates/events/customer/customer-created.event.ts:
Separar los archivos de eventos en las carpetas de sus entidades respectivas para mantener el orden

GerardoFdez7 commented on src/domains/customer/application/events/customer/identity-created.handler.ts:
```diff
@@ -0,0 +1,39 @@
+    if (event.auth.get('accountType').getValue() === AccountTypeEnum.CUSTOMER) {
+      // Use the full domain for tenant lookup
+      const fullDomain = event.domain;
+      const name = event.auth.get('email').getValue().split('@')[0];
```
Extraer esto en una función utilitaria en common/utils y utilizarla aquí y en el dominio del tenant que tambien se repite

GerardoFdez7 commented on src/domains/customer/application/mappers/review/customer-review-product-enriched.dto.ts:
```diff
@@ -0,0 +1,24 @@
+ * Data Transfer Object for CustomerReviewProduct enriched with variant details
+ * Used for API responses when retrieving reviews with product information
+ */
+export interface CustomerReviewProductWithVariantDTO
```
Este tipo es exactamente VariantDetailsDTO ubicado en /shared, no se si es estrictamente necesario, pero de serlo pienso que es mejor colocarlo en `customer-review-product.dto`

GerardoFdez7 commented on src/domains/customer/application/mappers/review/customer-review-product.mapper.ts:
```diff
@@ -0,0 +1,105 @@
+/**
+ * Interface for persistence CustomerReviewProduct model
+ */
+export interface ICustomerReviewProductPersistence {
```
Ya tienes estas interfaces exportadas en tu Value Object, no las repitas

GerardoFdez7 commented on src/domains/customer/application/mappers/wish-list/wish-list.mapper.ts:
```diff
@@ -0,0 +1,60 @@
+/**
+ * Interface for persistence WishList model
+ */
+export interface IWishListPersistence {
```
Ya tienes estas interfaces exportadas en tu Value Object, no las repitas

GerardoFdez7 commented on src/domains/customer/application/mappers/customer/customer.mapper.ts:
```diff
@@ -0,0 +1,114 @@
+/**
+ * Interface for persistence Customer model
+ */
+export interface ICustomerPersistence {
```
Ya tienes esta interface en customer.attributes

GerardoFdez7 commented on src/domains/customer/infrastructure/database/postgres/customer.repository.ts:
```diff
@@ -0,0 +1,180 @@
+        customer.get('defaultBillingAddressId')?.getValue() || null,
+      updatedAt: customer.get('updatedAt'),
+      createdAt: customer.get('createdAt'),
+    };
```
Ya tienes un mapper definido `DRY`

GerardoFdez7 commented on src/domains/cart/application/events/customer-created.handler.ts:
```diff
@@ -1,6 +1,6 @@
 import { EventsHandler, IEventHandler, CommandBus } from '@nestjs/cqrs';
 import { Injectable } from '@nestjs/common';
-import { CustomerCreatedEvent } from '../../aggregates/events/customer-created.event'; // replace with the correct path '@customer/aggregates/events'
+import { CustomerCreatedEvent } from '../../../customer/aggregates/events/customer-created.event'; // replace with the correct path '@customer/aggregates/events'
```
Ya puedes eliminar el comentario

GerardoFdez7 commented on src/domains/customer/aggregates/entities/index.ts:
`Export *` puede causar errores dependencia circular, exporta explicitamente lo que vas a usar.

GerardoFdez7 commented on src/domains/customer/aggregates/value-objects/index.ts:
Exportar CustomerReviewProduct

GerardoFdez7 commented on src/domains/customer/aggregates/value-objects/wish-list-item.vo.ts:
```diff
@@ -0,0 +1,93 @@
+  variantId: z.string().uuid(),
+  customerId: z.string().uuid(),
+});
+
```
Usar uuidv7 from zod/v4 o si es posible dentro del object reemplazar con el vo Id directamente para no repetir la definicion de Id

GerardoFdez7 commented on src/domains/tenant/aggregates/repositories/tenant.interface.ts:
```diff
@@ -40,4 +41,6 @@ export interface ITenantRepository {
    */
   findById(id: Id): Promise<Tenant | null>;
+
+  getTenantIdByDomain(domain: Domain): Promise<string | null>;
```
Agregar JSDoc

GerardoFdez7 commented on src/domains/tenant/infrastructure/persistence/postgres/tenant.repository.ts:
```diff
@@ -12,6 +12,7 @@ import { Tenant, ITenantType } from '../../../aggregates/entities';
 import { Prisma, Tenant as PrismaTenant } from '.prisma/postgres';
 import { ITenantRepository } from '../../../aggregates/repositories/tenant.interface';
 import { PrismaErrorUtils } from '@utils/prisma-error-utils';
+import { Domain } from 'src/domains/tenant/aggregates/value-objects';
```
Eliminar linea 9 y usar `import { Id, Domain } from ../../../aggregates/value-objects` porque esa forma con src ya va a estar deprecada y te va a dar error cuando hagas rebase

GerardoFdez7 commented on src/domains/customer/aggregates/repositories/wish-list.interface.ts:
```diff
@@ -0,0 +1,19 @@
+  removeVariantFromWishList(
+    customerId: Id,
+    variantId: Id,
+  ): Promise<WishListItem>;
```
Comentario generado por GitHub Copilot: The return type of removeVariantFromWishList is Promise<WishListItem>, but it is unclear what happens if the item does not exist. Consider documenting the behavior or returning Promise<WishListItem | null>.

GerardoFdez7 commented on src/domains/customer/aggregates/repositories/wish-list.interface.ts:
```diff
@@ -0,0 +1,19 @@
+    variantIds: Id[],
+  ): Promise<WishListItem[]>;
+  getManyWishListsByVariantIds(
+    variantsIds: Id[],
```
Comentario generado por GitHub Copilot: The parameter name variantsIds should be variantIds for consistency with the rest of the interface and to avoid confusion.

Suggested change:

-     variantsIds: Id[],
+     variantIds: Id[],

GerardoFdez7 commented on src/domains/customer/aggregates/repositories/wish-list.interface.ts:
```diff
@@ -0,0 +1,19 @@
+import { Id } from '@shared/value-objects';
+import { WishListItem } from '../value-objects';
+
+export interface IWishListRepository {
```
Falta JSDoc

GerardoFdez7 commented on src/domains/customer/aggregates/repositories/customer.interface.ts:
```diff
@@ -13,4 +14,10 @@ export interface ICustomerRepository {
   findByAuthIdentityId(
     authIdentityId: Id,
   ): Promise<{ id: string; tenantId: string } | null>;
+
+  create(customer: Customer): Promise<Customer>;
+
+  findCustomerById(id: Id, tenantId: Id): Promise<Customer>;
+
+  update(customer: Customer): Promise<Customer>;
 }
```
Agregar JSDoc

GerardoFdez7 commented on src/domains/customer/aggregates/repositories/customer.interface.ts:
```diff
@@ -13,4 +14,10 @@ export interface ICustomerRepository {
+
+  findCustomerById(id: Id, tenantId: Id): Promise<Customer>;
+
+  update(customer: Customer): Promise<Customer>;
```
Agregar tenantId para optimizar la busqueda

GerardoFdez7 commented on src/domains/customer/aggregates/repositories/customer.interface.ts:
```diff
@@ -13,4 +14,10 @@ export interface ICustomerRepository {
   findByAuthIdentityId(
```
Comentario generado por GitHub Copilot: The return type of findByAuthIdentityId is a plain object, while other methods return a Customer entity. Consider returning a Customer or null for consistency, unless there is a specific reason to return a subset.

Suggested change:
```
-   ): Promise<{ id: string; tenantId: string } | null>;
+   ): Promise<Customer | null>;
```

jruiz002 replied:
No es relevante.

GerardoFdez7 commented on src/domains/customer/aggregates/repositories/customer.interface.ts:
```diff
@@ -13,4 +14,10 @@ export interface ICustomerRepository {
+
+  create(customer: Customer): Promise<Customer>;
+
+  findCustomerById(id: Id, tenantId: Id): Promise<Customer>;
```
1. Quizas habria que agregar null a la Promise. 
2. Comentario generado por GitHub Copilot: The method name findCustomerById is inconsistent with the shorter name findByAuthIdentityId. Consider renaming to findById for consistency.

Suggested change:
```
-   findCustomerById(id: Id, tenantId: Id): Promise<Customer>;
+   findById(id: Id, tenantId: Id): Promise<Customer>;
```

GerardoFdez7 commented on src/domains/customer/aggregates/repositories/customer-review-product.interface.ts:
```diff
@@ -0,0 +1,33 @@
+import { Id } from '@shared/value-objects';
+import { CustomerReviewProduct } from '../value-objects/customer-review-product.vo';
```
Importar desde `../value-objects/`

GerardoFdez7 commented on src/domains/customer/aggregates/repositories/customer-review-product.interface.ts:
```diff
@@ -0,0 +1,33 @@
+   * @returns Promise that resolves to the CustomerReviewProduct if found, null otherwise
+   * @throws {Error} When repository operation fails
+   */
+  findById(id: Id): Promise<CustomerReviewProduct | null>;
```
Incluir customerId como parametro para optimizar la busqueda

GerardoFdez7 commented on src/domains/customer/aggregates/repositories/customer-review-product.interface.ts:
```diff
@@ -0,0 +1,33 @@
+   * @returns Promise that resolves to an array of CustomerReviewProduct entities
+   * @throws {Error} When repository operation fails
+   */
+  findMany(customerId: Id, reviewIds?: Id[]): Promise<CustomerReviewProduct[]>;
```
`reviewIds?: Id[]` Es opcional para que sirva de alguna manera hibrida? O sea que si reviewIds no se recibe entonces devuelve todas las reviews de un customer, y si se recibe entonces solo busca esas? De esa manera tambien se podría rehusar la query para los adapters externos al dominio customer si llegase a ser necesario en el futuro.

GerardoFdez7 commented on src/domains/customer/aggregates/value-objects/customer-review-product.vo.ts:
```diff
@@ -0,0 +1,119 @@
+  ratingCount: z.number().min(1).max(5),
+  comment: z.string().min(1),
+  customerId: z.uuidv7({ message: 'Id must be a valid UUID' }),
+  variantId: z.uuidv7({ message: 'Id must be a valid UUID' }),
```
Intentar rehusar Id vo

GerardoFdez7 commented on src/domains/customer/aggregates/value-objects/customer-review-product.vo.ts:
```diff
@@ -0,0 +1,119 @@
+
+  static create(props: CustomerReviewProductProps): CustomerReviewProduct {
+    customerReviewProductSchema.parse(props);
+    return new CustomerReviewProduct(props);
+  }
+
+  static fromPersistence(
+    props: CustomerReviewProductPropsWithId,
+  ): CustomerReviewProduct {
+    customerReviewProductSchema.parse(props);
+    return new CustomerReviewProduct(props, props.id, props.updatedAt);
```
Comentario generado por GitHub Copilot: customerReviewProductSchema.parse(props); will fail if props contains id and updatedAt fields, as the schema does not allow these. Consider omitting these fields before validation or extending the schema for persistence.

Suggested change:
```
-     customerReviewProductSchema.parse(props);
+     customerReviewProductSchema.parse({
+       ratingCount: props.ratingCount,
+       comment: props.comment,
+       customerId: props.customerId,
+       variantId: props.variantId,
+     });
```

GerardoFdez7 commented on src/domains/customer/aggregates/value-objects/customer-review-product.vo.ts:
```diff
@@ -0,0 +1,119 @@
+      updatedProps,
+      existingReview.getIdValue(),
+      new Date(), // Updated timestamp
+    );
```
Comentario generado por GitHub Copilot: Same as above, if updatedProps ever includes extra fields, validation will fail. Ensure only the expected fields are present.

GerardoFdez7 commented on src/domains/customer/aggregates/value-objects/customer-review-product.vo.ts:
```diff
@@ -0,0 +1,119 @@
+  ) {
+    this.id = existingId ? Id.create(existingId) : Id.generate();
+    this.ratingCount = props.ratingCount;
+    this.comment = LongDescription.create(props.comment);
```
Comentario generado por GitHub Copilot: LongDescription.create(props.comment) is called after schema validation, but if LongDescription.create also validates, this results in duplicate validation. Consider validating only once for efficiency.

Suggested change:

-   comment: z.string().min(1),
+   comment: z.string(), // Remove min(1) to avoid duplicate validation

GerardoFdez7 commented on src/domains/customer/aggregates/value-objects/wish-list-item.vo.ts:
```diff
@@ -0,0 +1,93 @@
+    return new WishListItem(props);
+  }
+
+  static fromPersistence(props: WishListPropsWithId): WishListItem {
+    wishListItemSchema.parse(props);
+    return new WishListItem(props, props.id, props.updatedAt);
+  }
```
Comentario generado por GitHub Copilot: The wishListItemSchema does not validate the id and updatedAt fields present in WishListPropsWithId. This means fromPersistence may accept invalid data for these fields.

Suggested change:

```
+ const wishListItemWithIdSchema = wishListItemSchema.extend({
+   id: z.string().uuid(),
+   updatedAt: z.coerce.date(),
+ });
+ 
-     wishListItemSchema.parse(props);
+     wishListItemWithIdSchema.parse(props);
```

GerardoFdez7 commented on src/domains/customer/aggregates/value-objects/wish-list-item.vo.ts:
```diff
@@ -0,0 +1,93 @@
+  public getCustomerIdValue(): string {
+    return this.customerId.getValue();
+  }
+
+  // Equals method for comparison
+  public equals(other: WishListItem): boolean {
+    return (
+      this.id.equals(other.id) &&
+      this.variantId.equals(other.variantId) &&
+      this.customerId.equals(other.customerId)
+    );
+  }
```
Comentario generado por GitHub Copilot: The equals method only compares id, variantId, and customerId, but not updatedAt. If updatedAt is relevant for equality, it should be included or documented why it is omitted.

Suggested change:
```
-   // Equals method for comparison
-   public equals(other: WishListItem): boolean {
-     return (
-       this.id.equals(other.id) &&
-       this.variantId.equals(other.variantId) &&
-       this.customerId.equals(other.customerId)
+   // Equals method for comparison, including updatedAt
+   public equals(other: WishListItem): boolean {
+     return (
+       this.id.equals(other.id) &&
+       this.variantId.equals(other.variantId) &&
+       this.customerId.equals(other.customerId) &&
+       this.updatedAt.getTime() === other.updatedAt.getTime()
```

GerardoFdez7 commented on src/domains/customer/aggregates/entities/customer/customer.entity.ts:
```diff
@@ -0,0 +1,162 @@
+  createdAt: Date;
+}
+
+export class Customer extends Entity<ICustomerProps> {
```
Comentario generado por GitHub Copilot: The Customer class lacks a class-level doc comment describing its domain role and usage. Adding this would improve maintainability.

GerardoFdez7 commented on src/domains/customer/aggregates/entities/customer/customer.entity.ts:
```diff
@@ -0,0 +1,162 @@
+
+    return customer;
+  }
+
+  static update(
+    customer: Customer,
+    input: Partial<Omit<ICustomerCreate, 'tenantId' | 'authIdentityId'>>,
+  ): Customer {
+    const customerUpdated = new Customer({
+      id: customer.get('id'),
+      name: input.name ? Name.create(input.name) : customer.get('name'),
+      tenantId: customer.get('tenantId'),
+      authIdentityId: customer.get('authIdentityId'),
+      defaultPhoneNumberId: input.defaultPhoneNumberId
+        ? Id.create(input.defaultPhoneNumberId)
+        : customer.get('defaultPhoneNumberId'),
+      defaultShippingAddressId: input.defaultShippingAddressId
+        ? Id.create(input.defaultShippingAddressId)
+        : customer.get('defaultShippingAddressId'),
+      defaultBillingAddressId: input.defaultBillingAddressId
+        ? Id.create(input.defaultBillingAddressId)
+        : customer.get('defaultBillingAddressId'),
+      updatedAt: new Date(),
+      createdAt: customer.get('createdAt'),
+    });
+
+    customerUpdated.apply(new CustomerUpdatedEvent(customerUpdated));
+
+    return customerUpdated;
+  }
```
Manten los metodos update como en los otros dominios:
```
 /**
   * Updates an existing Warehouse with new values
   * @param warehouse The existing Warehouse to update
   * @param updates The properties to update
   * @returns The updated Warehouse domain entity
   */
  static update(
    warehouse: Warehouse,
    updates: Partial<Omit<IWarehouseBase, 'tenantId'>>,
  ): Warehouse {
    const props = { ...warehouse.props };

    if (updates.name !== undefined) {
      props.name = Name.create(updates.name);
    }

    if (updates.addressId !== undefined) {
      props.addressId = Id.create(updates.addressId);
    }

    props.updatedAt = new Date();

    const updatedWarehouse = new Warehouse(props);

    // Apply domain event
    updatedWarehouse.apply(new WarehouseUpdatedEvent(updatedWarehouse));

    return updatedWarehouse;
  }
  ```

jruiz002 replied:
por qué jaja? hay mas de una forma de hacer la actualización y hace exactamente lo mismo, no afecta en nada.

GerardoFdez7 commented on src/domains/customer/aggregates/entities/customer/customer.entity.ts:
```diff
@@ -0,0 +1,162 @@
+    >,
+    customer: Customer,
+  ): CustomerReviewProduct {
+    const updatedReview = CustomerReviewProduct.update(existingReview, updates);
```
Comentario generado por GitHub Copilot: In updateCustomerReviewProduct, you call CustomerReviewProduct.update(existingReview, updates), but the type of updates is Partial<Pick<CustomerReviewProductPropsWithId, 'ratingCount' | 'comment'>>, while the update method expects Partial<Pick<CustomerReviewProductProps, 'ratingCount' | 'comment'>>. If CustomerReviewProductPropsWithId and CustomerReviewProductProps diverge, this could cause subtle bugs. Consider aligning the types for clarity and safety.

GerardoFdez7 commented on src/domains/customer/application/commands/delete/review/delete-customer-review-product.handler.ts:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    // Use the domain method to emit the event
+    Customer.removeCustomerReviewProduct(reviewFound, customerFound);
```
Comentario generado por GitHub Copilot: The repository method removeReview already checks if the review belongs to the customer and throws if not. The earlier manual check (lines 48–52) is redundant and could be removed to avoid double validation and keep logic DRY.

Suggested change:
```
-     // Validate that the review belongs to the customer
-     if (reviewFound.getCustomerIdValue() !== command.customerId) {
-       throw new NotFoundException(
-         `Review with ID ${command.reviewId} does not belong to customer ${command.customerId}`,
-       );
-     }
+     // The repository method removeReview will validate review ownership and throw if necessary
```

GerardoFdez7 commented on src/domains/customer/application/commands/delete/review/delete-customer-review-product.handler.ts:
```diff
@@ -0,0 +1,65 @@
+    // Find the review to validate it exists and belongs to the customer
+    const reviewFound = await this.reviewRepository.findById(reviewId);
+
+    if (!reviewFound) {
```
Comentario generado por GitHub Copilot: The static method Customer.removeCustomerReviewProduct is called directly, which may be confusing. Consider invoking this on the customer instance for clarity, e.g., customerFound.removeCustomerReviewProduct(reviewFound), if the domain model allows.

Suggested change:
```
-     Customer.removeCustomerReviewProduct(reviewFound, customerFound);
+     customerFound.removeCustomerReviewProduct(reviewFound);
```

jruiz002 replied:
No es necesario realemente jaja

GerardoFdez7 commented on src/domains/customer/application/commands/delete/wish-list/many/delete-many-wish-list.dto.ts:
Separar operaciones many y one en carpetas

jruiz002 replied:
ya lo hiciste tu por lo que veo jaja

GerardoFdez7 commented on src/domains/customer/application/commands/update/review/update-customer-review-product.handler.ts:
```diff
@@ -0,0 +1,60 @@
+    );
+
+    // Search review
+    const existingReview = await this.reviewRepository.findById(reviewId);
```
Comentario generado por GitHub Copilot: The result of findById can be null, but existingReview is passed to Customer.updateCustomerReviewProduct without a null check. This can cause a runtime error if the review does not exist.

GerardoFdez7 commented on src/domains/customer/application/commands/update/review/update-customer-review-product.handler.ts:
```diff
@@ -0,0 +1,60 @@
+    const reviewId = Id.create(command.review.id);
+
+    // Search customer
+    const customerFound = await this.customerRepository.findCustomerById(
```
Enviar una excepcion si el customer no se encuentra

GerardoFdez7 commented on src/domains/customer/application/mappers/customer/customer.dto.ts:
```diff
@@ -0,0 +1,24 @@
+ * Data Transfer Object for Customer entity
+ * Used for API responses when creating or retrieving customers
+ */
+export interface CustomerDTO {
+  id: string;
+  name: string;
+  tenantId: string;
+  authIdentityId: string;
+  defaultPhoneNumberId?: string | null;
+  defaultShippingAddressId?: string | null;
+  defaultBillingAddressId?: string | null;
+  updatedAt: Date;
+  createdAt: Date;
+}
```
Esto deberia ser igual a una de las interfaces que definiste en customer.attribues para evitar repetir los tipos.

GerardoFdez7 commented on src/domains/customer/application/mappers/customer/index.ts:
```diff
@@ -0,0 +1,2 @@
+export * from './customer.dto';
+export * from './customer.mapper';
```
`Export *` puede provocar errores de dependencias circulares

GerardoFdez7 commented on src/domains/customer/application/mappers/review/customer-review-product.dto.ts:
```diff
@@ -0,0 +1,21 @@
+ * Data Transfer Object for CustomerReviewProduct entity
+ * Used for API responses when creating or retrieving customer product reviews
+ */
+export interface CustomerReviewProductDTO {
```
Esto deberia ser igual a la interface que definiste en tu vo para no repetir lo mismo.

GerardoFdez7 commented on src/domains/customer/application/mappers/index.ts:
```diff
@@ -0,0 +1 @@
+export * from './customer';
```
Export * puede provocar errores de dependencias circulares

GerardoFdez7 commented on src/domains/customer/application/mappers/index.ts:
Exportar todos los mappers y dtos aqui

GerardoFdez7 commented on src/domains/customer/application/mappers/wish-list/wish-list.dto.ts:
```diff
@@ -0,0 +1,32 @@
+ * Data Transfer Object for WishList entity
+ * Used for API responses when creating or retrieving wishlist items
+ */
+export interface WishListDTO {
```
Esto deberia ser igual a la interface que definiste en tu vo para no repetir lo mismo.

GerardoFdez7 commented on src/domains/customer/application/mappers/wish-list/index.ts:
Eliminar este index y solo usar el que esta dentro de mappers

GerardoFdez7 commented on src/domains/customer/application/mappers/customer/index.ts:
Eliminar este index y solo usar el que esta dentro de mappers

GerardoFdez7 commented on src/domains/customer/application/queries/index.ts:
Exportar todos los dtos y queries handlers para definir los imports de forma limpia en el module y en el resolver

GerardoFdez7 commented on src/domains/customer/application/queries/many/review/find-many-customer-reviews.dto.ts:
```diff
@@ -0,0 +1,8 @@
+    public readonly customerId: string,
+    public readonly reviewIds?: string[],
+    public readonly page: number = 1,
+    public readonly limit: number = 25,
```
Si ya tienes estos valores inicializados en tu resolver no es necesario hacerlo aqui tambien

GerardoFdez7 commented on src/domains/customer/application/queries/many/wish-list/find-wish-list-items.dto.ts:
```diff
@@ -0,0 +1,8 @@
+    public readonly customerId: string,
+    public readonly variantsIds: string[] = [],
+    public readonly page: number = 1,
+    public readonly limit: number = 25,
```
Si ya tienes estos valores inicializados en tu resolver no es necesario hacerlo aqui tambien

GerardoFdez7 commented on src/domains/customer/application/queries/many/review/find-many-customer-reviews.handler.ts:
```diff
@@ -0,0 +1,91 @@
+    const offset = (page - 1) * limit;
+
+    // Slice items for requested page
+    const pagedReviews = reviews.slice(offset, offset + limit);
```
Comentario generado por GitHub Copilot: Slicing the entire reviews array after fetching all records from the repository can be inefficient for large datasets. Prefer paginating at the database query level if possible.

Suggested change:
```
-     // Get reviews from repository
-     const reviews = await this.customerReviewProductRepository.findMany(
-       customerId,
-       reviewIds,
-     );
- 
-     const total = reviews.length;
- 
-     if (total === 0) {
-       return { reviews: [], total: 0, hasMore: false };
-     }
- 
-     // Normalize pagination params
-     const page = Math.max(1, query.page ?? 1);
-     const limit = Math.min(50, Math.max(1, query.limit ?? 25));
-     const offset = (page - 1) * limit;
- 
-     // Slice items for requested page
-     const pagedReviews = reviews.slice(offset, offset + limit);
+     // Normalize pagination params
+     const page = Math.max(1, query.page ?? 1);
+     const limit = Math.min(50, Math.max(1, query.limit ?? 25));
+     const offset = (page - 1) * limit;
+ 
+     // Get paginated reviews and total count from repository
+     const { reviews: pagedReviews, total } = await this.customerReviewProductRepository.findManyPaginated(
+       customerId,
+       reviewIds,
+       offset,
+       limit,
+     );
+ 
+     if (total === 0) {
+       return { reviews: [], total: 0, hasMore: false };
+     }
```

Basicamente que no es necesario hacer la paginacion desde aqui si la puedes hacer desde la db pasando los parametros al metodo en el repositorio

GerardoFdez7 commented on src/domains/customer/application/queries/many/wish-list/find-wish-list-items.handler.ts:
```diff
@@ -0,0 +1,83 @@
+
+  async execute(query: FindWishlistItemsDto): Promise<PaginatedWishlistDTO> {
+    const customerId = Id.create(query.customerId);
+    const variantIds = query.variantsIds.map((id) => Id.create(id));
```
Comentario generado por GitHub Copilot: No null/undefined check for query.variantsIds. If variantsIds is missing or not an array, .map will throw. Consider validating or defaulting to an empty array.

Suggested change:
```
-     const variantIds = query.variantsIds.map((id) => Id.create(id));
+     const variantIds = Array.isArray(query.variantsIds)
+       ? query.variantsIds.map((id) => Id.create(id))
+       : [];
```

GerardoFdez7 commented on src/domains/customer/application/queries/many/wish-list/find-wish-list-items.handler.ts:
```diff
@@ -0,0 +1,83 @@
+    const pagedItems = wishListItems.slice(offset, offset + limit);
+
+    // Extract variant IDs from paged wishlist items to get variant details
+    const variantIdStrings = pagedItems.map((item) => item.getVariantIdValue());
```
Lo mismo, no es necesario hacer la paginacion desde aqui si la puedes hacer desde la db pasando los parametros al metodo en el repositorio

GerardoFdez7 commented on src/domains/customer/presentation/graphql/types/customer.types.ts:
```diff
@@ -0,0 +1,274 @@
+  @Field(() => Float)
+  price: number;
+
+  @Field(() => Boolean, { nullable: true })
```
Comentario generado por GitHub Copilot: The property isArchived is marked as nullable in the GraphQL schema but is not optional in the TypeScript type. Consider making it optional (isArchived?: boolean) for consistency.

Suggested change:

-   isArchived: boolean;
+   isArchived?: boolean;

GerardoFdez7 commented on src/domains/customer/presentation/graphql/types/customer.types.ts:
```diff
@@ -0,0 +1,274 @@
+  @Field(() => Float)
+  price: number;
+
+  @Field(() => Boolean, { nullable: true })
```
Comentario generado por GitHub Copilot: The property isArchived is marked as nullable in the GraphQL schema but is not optional in the TypeScript type. Consider making it optional (isArchived?: boolean) for consistency.

Suggested change:

-   isArchived: boolean;
+   isArchived?: boolean;

GerardoFdez7 commented on src/domains/customer/presentation/graphql/customer.resolver.ts:
```diff
@@ -0,0 +1,200 @@
+import { UpdateCustomerReviewProductDto } from '../../application/commands/update/review/update-customer-review-product.dto';
+import { DeleteCustomerReviewProductDto } from '../../application/commands/delete/review/delete-customer-review-product.dto';
+import { FindManyCustomerReviewsDto } from '../../application/queries/many/review/find-many-customer-reviews.dto';
+import { UpdateCustomerDto } from '../../application/commands/update/customer/update-customer.dto';
```
Juntar imports segun index

GerardoFdez7 commented on src/domains/customer/presentation/graphql/customer.resolver.ts:
```diff
@@ -0,0 +1,200 @@
+  @Query(() => PaginatedWishlistType)
+  async getWishListItems(
+    @Args('input', { type: () => GetWishlistPaginatedInput })
+    input: GetWishlistPaginatedInput,
```
Pienso que para mantener la consistencia con todos los dominios deberías sacar esos inputs de ese objeto y exponerlos de forma normal

GerardoFdez7 commented on src/domains/customer/presentation/graphql/customer.resolver.ts:
```diff
@@ -0,0 +1,200 @@
+  @Query(() => PaginatedCustomerReviewProductWithVariantType)
+  async getCustomerReviews(
+    @Args('input', { type: () => GetCustomerReviewsPaginatedInput })
+    input: GetCustomerReviewsPaginatedInput,
```
Pienso que para mantener la consistencia con todos los dominios deberías sacar esos inputs de ese objeto y exponerlos de forma normal

GerardoFdez7 commented on src/domains/customer/presentation/graphql/customer.resolver.ts:
```diff
@@ -0,0 +1,200 @@
+    @CurrentUser() user: JwtPayload,
+  ): Promise<boolean> {
+    await this.commandBus.execute(
+      new DeleteCustomerReviewProductDto(
+        user.customerId,
+        input.id,
+        user.tenantId,
+      ),
+    );
+    return true;
+  }
+
+  @Mutation(() => Boolean)
+  async removeVariantFromWishList(
+    @Args('input', { type: () => WishListItemDeleteInput })
+    input: WishListItemDeleteInput,
+    @CurrentUser() user: JwtPayload,
+  ): Promise<boolean> {
+    await this.commandBus.execute(
+      new DeleteWishListDto(user.customerId, input.variantId, user.tenantId),
+    );
+    return true;
+  }
+
+  @Mutation(() => Boolean)
+  async removeManyVariantsFromWishList(
+    @Args('input', { type: () => WishListManyItemsInput })
+    input: WishListManyItemsInput,
+    @CurrentUser() user: JwtPayload,
+  ): Promise<boolean> {
+    await this.commandBus.execute(
+      new DeleteManyWishListDto(
+        user.customerId,
+        input.variantIds,
+        user.tenantId,
+      ),
+    );
+    return true;
+  }
+
```
Usa el spread operator en el input para no volver a definir lo que ya definiste que quieres en el input object y en el dto:
```
  @Mutation(() => CategoryType)
  async createCategory(
    @Args('input', { type: () => CreateCategoryInput })
    input: CreateCategoryInput,
    @CurrentUser() user: JwtPayload,
  ): Promise<CategoryType> {
    const inputWithTenantId = { ...input, tenantId: user.tenantId };
    return this.commandBus.execute(new CreateCategoryDTO(inputWithTenantId));
  } 
```

GerardoFdez7 commented on src/domains/customer/infrastructure/persistence/postgres/customer-review-product.repository.ts:
```diff
@@ -0,0 +1,227 @@
+   * @param reviewId The unique identifier of the review product.
+   * @returns Promise that resolves when the review is successfully removed.
+   */
+  async removeReview(customerId: Id, reviewId: Id): Promise<void> {
```
Comentario generado por GitHub Copilot: The Prisma delete method only accepts a unique identifier for the where clause. Passing both id and customerId will throw an error. You should use deleteMany if you want to match both fields.

Suggested change:
```
-       const deletedReview =
-         await this.postgresService.customerReviewProduct.delete({
-           where: {
-             id: reviewIdValue,
-             customerId: customerIdValue,
-           },
-         });
- 
-       if (!deletedReview) {
+       const deleteResult =
+         await this.postgresService.customerReviewProduct.deleteMany({
+           where: {
+             id: reviewIdValue,
+             customerId: customerIdValue,
+           },
+         });
+ 
+       if (deleteResult.count === 0) {
```

GerardoFdez7 commented on src/domains/customer/infrastructure/database/postgres/customer-review-product.repository.ts:
```diff
@@ -0,0 +1,227 @@
+          },
+        });
+
+      if (!deletedReview) {
```
Comentario generado por GitHub Copilot: The Prisma delete method will throw if the record is not found, so the check for !deletedReview is redundant and unreachable.

Suggested change:
```
-       const deletedReview =
-         await this.postgresService.customerReviewProduct.delete({
-           where: {
-             id: reviewIdValue,
-             customerId: customerIdValue,
-           },
-         });
- 
-       if (!deletedReview) {
-         throw new DatabaseOperationError(
-           'remove customer review product',
-           'Customer review product not found or does not belong to the specified customer',
-         );
-       }
+       await this.postgresService.customerReviewProduct.delete({
+         where: {
+           id: reviewIdValue,
+           customerId: customerIdValue,
+         },
+       });
```

GerardoFdez7 commented on src/domains/customer/infrastructure/persistence/postgres/customer-review-product.repository.ts:
```diff
@@ -0,0 +1,227 @@
+  /**
+   * Centralized error handling for database operations
+   */
+  private handleDatabaseError(error: unknown, operation: string): never {
```
Comentario generado por GitHub Copilot: The handleDatabaseError method is marked as private but is used as a return value in several catch blocks. Consider always throwing from this method and not returning it.

Suggested change:
```
-       return this.handleDatabaseError(error, 'create customer review product');
+       this.handleDatabaseError(error, 'create customer review product');
-       return this.handleDatabaseError(error, 'update customer review product');
+       this.handleDatabaseError(error, 'update customer review product');
-       return this.handleDatabaseError(
+       this.handleDatabaseError(
-       return this.handleDatabaseError(
+       this.handleDatabaseError(
-       return this.handleDatabaseError(error, 'remove customer review product');
+       this.handleDatabaseError(error, 'remove customer review product');
```

GerardoFdez7 commented on src/domains/customer/infrastructure/database/postgres/customer-review-product.repository.ts:
```diff
@@ -0,0 +1,227 @@
+import { DatabaseOperationError } from '@shared/errors';
+import { Id } from '@shared/value-objects';
+import { ICustomerReviewProductRepository } from 'src/domains/customer/aggregates/repositories/customer-review-product.interface';
+import { CustomerReviewProduct } from 'src/domains/customer/aggregates/value-objects/customer-review-product.vo';
```
Usar src te va a dar error cuando hagas rebase

GerardoFdez7 reviewed (changes requested):
Ya casi esta

jruiz002 committed:
refactor(customer): reorganize event files into domain-specific directories

Move customer domain event files into their respective domain directories (customer, wishlist, review) for better organization
Remove duplicate handler registrations in customer module
Update imports to use new file paths

jruiz002 committed:
refactor(customer): reorganize mappers and improve type safety

- remove separate mapper index files and consolidate exports
- replace persistence interfaces with domain types
- update zod schema to use uuidv7
- make pagination params optional in DTOs
- add documentation for repository methods

GerardoFdez7 committed:
test(customer): add integration tests for the create and update operations

GerardoFdez7 committed:
refactor(zod): replace uuidv7 with uuid

jruiz002 committed:
fix(graphql): make isArchived field nullable in types

refactor(repository): update import path to use relative path

jruiz002 committed:
refactor: update import paths to use relative paths instead of absolute

fix: update zod import to use zod/v4 in value objects
docs: remove duplicate comment in graphql schema
style: clean up graphql schema by removing duplicate mutation

jruiz002 committed:
docs(customer): add jsdoc comments to customer repository interface methods

GerardoFdez7 committed:
test(review): add integration tests for customer review product commands

- Implement tests for creating customer review products, ensuring proper handling of customer retrieval, review creation, and event publishing.
- Add tests for deleting customer review products, focusing on ownership validation, review removal, and error handling.
- Create tests for updating customer review products, validating customer and review retrieval, processing updates, and ensuring correct event handling.
- Cover edge cases and error scenarios across all command tests to ensure robustness and reliability.

GerardoFdez7 committed:
test(wish-list): add integration tests for customer wish-list product commands

- Add comprehensive unit tests for all handlers to ensure correct functionality and error handling.
- Ensure proper integration with repositories and domain logic for wishlist item removal.

jruiz002 committed:
fix(wish-list): update return type to handle null case in removeVariant

Modify removeVariantFromWishList to return null instead of throwing when item not found
Fix typo in parameter name variantsIds to variantIds

jruiz002 committed:
feat(customer): add tenantId to customer update operation

Update customer repository interface and implementation to include tenantId in update queries

jruiz002 committed:
refactor(customer): rename findCustomerById to findById and allow null return

Update repository method name to be more concise and align with common naming conventions. Also modify the method to return null instead of throwing when customer is not found, making it more flexible for callers to handle missing cases.

jruiz002 committed:
fix(customer): add customerId to review product lookup

Ensure reviews are only accessible by their owners by including customerId in the findById query. This prevents customers from accessing reviews that don't belong to them.

jruiz002 committed:
fix(customer-review-product): validate only required fields in fromPersistence

Ensure only necessary fields are validated when creating from persistence to avoid potential validation errors with extra properties

jruiz002 committed:
refactor(customer-review): destructure props for schema validation clarity

jruiz002 committed:
refactor(customer): relax comment validation and extend wish list schema

remove minimum length requirement for customer review comments
add new schema for wish list items with id and updatedAt fields

jruiz002 committed:
fix(wish-list-item): add updatedAt comparison in equals method

jruiz002 committed:
refactor(customer): update customer entity type imports and add class docs

Improve type imports by using more generic CustomerReviewProductProps instead of CustomerReviewProductPropsWithId. Add comprehensive class documentation to clarify the Customer entity's role and responsibilities.

jruiz002 committed:
refactor(customer): move review ownership validation to repository

Remove duplicate validation in delete handler and add missing checks in update handler

jruiz002 committed:
refactor(customer-review): remove redundant return statements in error handling

The error handler already throws exceptions, so returning its result is unnecessary. This change simplifies the code by removing redundant return statements when calling handleDatabaseError.

jruiz002 committed:
refactor(customer): simplify review deletion by removing redundant check

The deleted review existence check is unnecessary since Prisma already handles not found cases with PrismaClientKnownRequestError. This makes the code cleaner while maintaining the same error handling behavior.

jruiz002 committed:
refactor(resolver): simplify dto creation by using object spread

Use object spread operator to simplify dto creation in resolver methods, reducing code duplication and improving maintainability

jruiz002 committed:
fix(wish-list): handle non-array variantsIds in wishlist query

Ensure the handler gracefully handles cases where variantsIds is not an array by defaulting to an empty array

jruiz002 committed:
refactor(graphql): simplify pagination inputs by removing dedicated input types

Move pagination parameters directly into resolver arguments for getWishListItems and getCustomerReviews queries to reduce boilerplate and improve maintainability

jruiz002 committed:
fix(customer-repository): make reviewIds parameter required and return nullable customer

Update method signatures to enforce required reviewIds parameter in findMany and allow null return from findById

GerardoFdez7 commented:
Fixing:
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @copilot-pull-request-reviewer: review summary — Pull Request Overview
- @GerardoFdez7: review summary — Ya casi esta
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Quality check failed!** Please fix the issues before merging this P…
- @coderabbitai: comment — [![Review Change Stack](https://storage.googleapis.com/coderabbit_publi…
- @github-actions: comment — ❌ **Quality check failed!** Please fix the issues before merging this P…
- @github-actions: comment — ❌ **Quality check failed!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @coderabbitai: review summary — Actionable comments posted: 14**

Will be in the next commit.

GerardoFdez7 commented:
Fixing:
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @copilot-pull-request-reviewer: review summary — Pull Request Overview
- @GerardoFdez7: review summary — Ya casi esta
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Quality check failed!** Please fix the issues before merging this P…
- @coderabbitai: comment — [![Review Change Stack](https://storage.googleapis.com/coderabbit_publi…
- @github-actions: comment — ❌ **Quality check failed!** Please fix the issues before merging this P…
- @github-actions: comment — ❌ **Quality check failed!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @coderabbitai: review summary — Actionable comments posted: 14**
- @GerardoFdez7: comment — Fixing:
- @coderabbitai: comment — [!TIP]
- @GerardoFdez7: comment — Fixing:
- @coderabbitai: comment — [!TIP]
- @GerardoFdez7: comment — Fixing:
- @coderabbitai: comment — [!TIP]

Will be in the next commit.

GerardoFdez7 commented:
Fixing:
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @copilot-pull-request-reviewer: review summary — Pull Request Overview
- @GerardoFdez7: review summary — Ya casi esta
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Quality check failed!** Please fix the issues before merging this P…
- @coderabbitai: comment — [![Review Change Stack](https://storage.googleapis.com/coderabbit_publi…
- @github-actions: comment — ❌ **Quality check failed!** Please fix the issues before merging this P…
- @github-actions: comment — ❌ **Quality check failed!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @coderabbitai: review summary — Actionable comments posted: 14**
- @GerardoFdez7: comment — Fixing:

Will be in the next commit.

GerardoFdez7 commented:
Fixing:
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @copilot-pull-request-reviewer: review summary — Pull Request Overview
- @GerardoFdez7: review summary — Ya casi esta
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Quality check failed!** Please fix the issues before merging this P…
- @coderabbitai: comment — [![Review Change Stack](https://storage.googleapis.com/coderabbit_publi…
- @github-actions: comment — ❌ **Quality check failed!** Please fix the issues before merging this P…
- @github-actions: comment — ❌ **Quality check failed!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @coderabbitai: review summary — Actionable comments posted: 14**
- @GerardoFdez7: comment — Fixing:
- @coderabbitai: comment — [!TIP]
- @GerardoFdez7: comment — Fixing:
- @coderabbitai: comment — [!TIP]

Will be in the next commit.

GerardoFdez7 commented:
Fixing:
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @copilot-pull-request-reviewer: review summary — Pull Request Overview
- @GerardoFdez7: review summary — Ya casi esta
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Build or tests failed!** Please fix the issues before merging this…
- @github-actions: comment — ❌ **Quality check failed!** Please fix the issues before merging this P…
- @coderabbitai: comment — [![Review Change Stack](https://storage.googleapis.com/coderabbit_publi…
- @github-actions: comment — ❌ **Quality check failed!** Please fix the issues before merging this P…
- @github-actions: comment — ❌ **Quality check failed!**
- @github-actions: comment — ✅ **Success! All checks have passed successfully!**
- @coderabbitai: review summary — Actionable comments posted: 14**
- @GerardoFdez7: comment — Fixing:
- @coderabbitai: comment — [!TIP]
- @GerardoFdez7: comment — Fixing:
- @coderabbitai: comment — [!TIP]
- @GerardoFdez7: comment — Fixing:
- @coderabbitai: comment — [!TIP]

Will be in the next commit.
